### PR TITLE
Add browser source interaction

### PR DESCRIPTION
**Pull request by EddyGharbi**

### CODE REVIEW

File: obs-studio-server/source/osn-source.cpp [hunk 1]
```diff
@@ -734,6 +775,131 @@ void osn::Source::SetEnabled(
...
+
+	if (src == nullptr) {
+		rval.push_back(ipc::value((uint64_t)ErrorCode::InvalidReference));
+		rval.push_back(ipc::value("Source reference is not valid."));
+		AUTO_DEBUG;
+		return;
+	}
+
+	char* text = new char[args[2].value_str.size()];
+	strcpy(text, args[2].value_str.c_str());
```
Comment (summeroff) [inline obs-studio-server/source/osn-source.cpp:889]: isnt it need to be size()+1 to copy zero character at the end of the c string?
can we send "args[2].value_str.c_str()" without creating new string?

File: obs-studio-server/source/osn-source.cpp [hunk 2]
```diff
@@ -154,6 +154,47 @@ void osn::Source::Register(ipc::server& srv)
...
+	                           ipc::type::Int32,
+	                           ipc::type::UInt32},
+	    SendMouseClick));
+	cls->register_function(std::make_shared<ipc::function>(
+	    "SendMouseMove",
+	    std::vector<ipc::type>{ipc::type::UInt64,
+	                           ipc::type::UInt32,
+	                           ipc::type::UInt32,
+	                           ipc::type::UInt32,
+                               ipc::type::Int32},
```
Comment (pdamasceno) [inline obs-studio-server/source/osn-source.cpp]: Indentation is different here

Comment (EddyGharbi) [reply obs-studio-server/source/osn-source.cpp]: Fixed, ty, good catch

File: obs-studio-server/source/osn-source.cpp (hunk 1 above)
Comment (EddyGharbi) [reply obs-studio-server/source/osn-source.cpp:889]: No `obs_key_event` only takes a `char*` not `const char*`
Fixing the size, good catch

**Review by summeroff — APPROVED** [review]

**Review by pdamasceno — APPROVED** [review]

**Pull request merged**